core/exception/receiver_spec.rb

```diff
@@ -2,13 +2,54 @@
 
 ruby_version_is "2.3" do
   describe "NameError#receiver" do
+    class ::ReceiverClass
+      def call_undefined_class_variable; @@doesnt_exist end
+    end
+
+    after :all do
+      Object.send(:remove_const, :ReceiverClass)
+    end
+
     it "returns the object that raised the exception" do
       receiver = Object.new
-      -> { receiver.instance_eval { doesnt_exist } }.should(
-        raise_error(NameError) do |error|
-          error.receiver.should equal(receiver)
-        end
-      )
+
+      begin
+        receiver.instance_eval { doesnt_exist }
```

Comment:
@eregon it is actually a question to @kachick since `instance_eval` was used in the first place. I think `receiver.doesnt_exist` should be good.